src/DebugEngineHost/HostNatvisProject.cs

```diff
@@ -73,6 +81,84 @@ internal enum VSENUMPROJFLAGS
                 //  file as well as all virtual projects including nested (a.k.a. sub) projects
             };
 
+            /// <summary>
+            /// Gets the WorkspaceService from Microsoft.VisualStudio.Workspace
+            /// </summary>
+            /// <remarks>This package won't be automatically loaded by MEF, so we need to manually acquire exported MEF Parts.</remarks>
+            private static Lazy<IVsFolderWorkspaceService> WorkspaceService
+            {
+                get
+                {
+                    Lazy<IVsFolderWorkspaceService> workspaceService = null;
+
+                    IComponentModel componentModel = ServiceProvider.GlobalProvider.GetService(typeof(SComponentModel).GUID) as IComponentModel;
+                    IEnumerable<Lazy<IVsFolderWorkspaceService>> workspaceServices = componentModel.DefaultExportProvider.GetExports<IVsFolderWorkspaceService>();
+
+                    if (workspaceServices != null && workspaceServices.Count() == 1)
+                    {
+                        workspaceService = new Lazy<IVsFolderWorkspaceService>(() =>
+                        {
+                            return workspaceServices.First().Value;
```

Comment:
could it export an an empty IEnumerable? If so then the First() call will throw.

Reply:
which is why the count is checked.

Reply:
Ah, missed that part. sorry!